### PR TITLE
[TT-6658] Implement getting mock response from OAS

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR implements getting mock response from OAS examples.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4318/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3151178520)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4318/merge
Commit: 8db79e8569527d903febcf51c2655d058701397c 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3151206654)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4318/merge
Commit: 3fc409312ff9300247a7b642e27b308b960e609e 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3151203249)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4318/merge
Commit: a61fe14a65d1f6ed2c20eeb8fcc322f0fe4ff41f 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3151218631)

File: apidef/oas/schema/x-tyk-gateway.md [hunk 1]
```diff
@@ -1116,4 +1116,27 @@
 
   **Type: `string`**
 
+- **`fromOASExamples`**
```
Comment (lghiur) [inline apidef/oas/schema/x-tyk-gateway.md:1119]: @furkansenharputlu can we add more description to these fields?

File: apidef/oas/operation.go [hunk 2]
```diff
@@ -474,10 +474,18 @@ func (s *OAS) fillOASValidateRequest(metas []apidef.ValidateRequestMeta) {
 }
 
 type MockResponse struct {
-	Enabled bool              `bson:"enabled" json:"enabled"`
-	Code    int               `bson:"code,omitempty" json:"code,omitempty"`
-	Body    string            `bson:"body,omitempty" json:"body,omitempty"`
-	Headers map[string]string `bson:"headers,omitempty" json:"headers,omitempty"`
+	Enabled         bool              `bson:"enabled" json:"enabled"`
```
Comment (jeffy-mathew) [inline apidef/oas/operation.go:477]: shall we add go doc comment for this?

File: gateway/mock_response.go [hunk 3]
```diff
@@ -57,3 +77,77 @@ func mockFromConfig(tykMockRespOp *oas.MockResponse) (int, []byte, map[string]st
...
+		return http.StatusNotFound, "", nil, nil, fmt.Errorf("there is no example response for the code: %d", code)
+	}
+
+	media := response.Value.Content.Get(contentType)
+	if media == nil {
+		return http.StatusNotFound, "", nil, nil, errors.New("there is no example response for the content type: " + contentType)
+	}
+
+	headers := make(map[string]string)
+	/*for key, val := range response.Value.Headers {
```
Comment (jeffy-mathew) [inline gateway/mock_response.go:119]: ?

File: gateway/mock_response.go [hunk 4]
```diff
@@ -57,3 +77,77 @@ func mockFromConfig(tykMockRespOp *oas.MockResponse) (int, []byte, map[string]st
...
+			}
+		} else {
+			for _, iterExample := range media.Examples {
+				example = iterExample.Value.Value
+				break
+			}
+		}
+	}
+
+	/*if example == nil {
```
Comment (jeffy-mathew) [inline gateway/mock_response.go:143]: ?

File: apidef/oas/operation.go (hunk 2 above)
Comment (furkansenharputlu) [reply apidef/oas/operation.go:477]: Yes I will add them

File: gateway/mock_response.go (hunk 3 above)
Comment (furkansenharputlu) [reply gateway/mock_response.go:119]: These comments will be active when I push extractor code.

**Review by jeffy-mathew — APPROVED** [review]

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4318/merge
Commit: fad901acdbc269cdd1166ced92b85ba36423638d 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3158456638)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: c5159ebc8167286b773518075e873333198f223f [TT-6658] Implement getting mock response from OAS (#4318)

* Implement getting mock response from OAS

* Add GoDoc and go generate
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3158531371)

**Pull request merged**